### PR TITLE
Change localized strings to computed

### LINKED ISSUE #527: Localized string constants are stored when they should be computed
Because a localized string is based on a locale and the locale can change while the app is running, localized string "constants" should be computed, not stored. In some places, like [this extension](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/db50212bb66caf94cf47e8f9f5ccb4643c553779/Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift#L791), the localized strings are stored as static properties, meaning that they won't change if that app's locale were to change. All static localized strings should be converted from stored to computed.

### PR DESCRIPTION
**Pull request by dfeinzimer**
Closes #527

### CODE REVIEW

File: Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift [hunk 1]
```diff
@@ -807,164 +807,213 @@ public struct UtilityNetworkTrace: View {
...
-        bundle: .toolkitModule,
-        comment: "A label indicating that no utility network trace configuration has been selected."
-    )
-    
-    static let resultsOptionLabel = String(
-        localized: "Results",
-        bundle: .toolkitModule,
-        comment: "A label for a button to show utility network trace results."
-    )
+        )}
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift:902]:
```suggestion
        )
    }
```

File: Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift [hunk 2]
```diff
@@ -807,164 +807,213 @@ public struct UtilityNetworkTrace: View {
...
-        bundle: .toolkitModule,
-        comment: """
+    static var featureResultsTitle: Self {
+        .init(
+            localized: "Feature Results",
+            bundle: .toolkitModule,
+            comment: """
                  A label in reference to utility elements returned as results of a utility network
                  trace operation.
                  """
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift:882]:
```suggestion
                A label in reference to utility elements returned as results of a utility network
                trace operation.
                """
```

File: Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift [hunk 3]
```diff
@@ -807,164 +807,213 @@ public struct UtilityNetworkTrace: View {
...
+    }
+    
+    static var functionResultsSectionTitle: Self {
+        .init(
+            localized: "Function Results",
+            bundle: .toolkitModule,
+            comment: """
                  A label in reference to function outputs returned as results of a utility network
                  trace operation.
                  """
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift:901]:
```suggestion
                A label in reference to function outputs returned as results of a utility network
                trace operation.
                """
```

File: Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift [hunk 4]
```diff
@@ -807,164 +807,213 @@ public struct UtilityNetworkTrace: View {
...
-        bundle: .toolkitModule,
-        comment: """
+    static var startingPointsTitle: Self {
+        .init(
+            localized: "Starting Points",
+            bundle: .toolkitModule,
+            comment: """
                  A label in reference to the utility elements chosen as starting points for a utility
                  network trace operation.
                  """
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift:968]:
```suggestion
                A label in reference to the utility elements chosen as starting points for a utility
                network trace operation.
                """
```

**Review by philium — COMMENTED** [review]
FYI: Several of the strings that represent button titles, such as "Add new", do not use title-style capitalization, [per the HIG](https://developer.apple.com/design/human-interface-guidelines/buttons#Content):

> image

**Review by philium — DISMISSED** [review]
Do the non-English translations for the old strings need to be removed?

**Comment by dfeinzimer** [comment]
> Do the non-English translations for the old strings need to be removed?

Yes, when we make the translation request towards the end of the release, the translation team should detect the changes and provide updated translations.

**Comment by dfeinzimer** [comment]
Updated component screenshots

**Review by mhdostal — APPROVED** [review]

**Pull request merged**